cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1738,16 +1768,61 @@ table_with_metadata parse_nested_json(host_span<SymbolT const> input,
   std::vector<column_name_info> out_column_names;
 
   // Iterate over the struct's child columns and convert to cudf column
+  size_type column_index = 0;
   for (auto const& col_name : root_struct_col.column_order) {
     auto const& json_col = root_struct_col.child_columns.find(col_name)->second;
     // Insert this columns name into the schema
     out_column_names.emplace_back(col_name);
 
+    std::optional<schema_element> child_schema_element = std::visit(
+      cudf::detail::visitor_overload{
+        [column_index](const std::vector<data_type>& user_dtypes) -> std::optional<schema_element> {
+          auto ret = (static_cast<std::size_t>(column_index) < user_dtypes.size())
+                       ? std::optional<schema_element>{{{}, user_dtypes[column_index]}}
+                       : std::optional<schema_element>{};
+#ifdef NJP_DEBUG_PRINT
+          std::cout << "Column by index: #" << column_index << ", type id: "
+                    << (ret.has_value() ? std::to_string(static_cast<int>(ret->type.id())) : "n/a")
+                    << ", with " << (ret.has_value() ? ret->child_types.size() : 0) << " children"
+                    << "\n";
+#endif
```

Comment:
Will these be removed?

Reply:
are the debug prints inconvenient while reading the code?
These debug prints help a lot during development and debugging.

Reply:
I find them in the way as a reader/reviewer. And most code is read more than written. 
Does this code need to be maintained? Should it be reviewed for correctness?
If yes to both then we should leave it. 
We should probably have rules for this since I'm seeing this show up in other code reviews.

Reply:
Agreed. I also find them hindering the reading flow but, as @karthikeyann pointed out, they also proofed to be really helpful for debugging. My plan and suggestion would be to keep them for a couple more weeks until we gained confidence that everything is working reliably and to remove them afterwards.